File: scalechain-storage/src/test/scala/io/scalechain/blockchain/storage/index/TransactingRocksDatabasePerformanceSpec.scala
package io.scalechain.blockchain.storage.index

import java.io.File

import io.scalechain.blockchain.storage.Storage
import org.apache.commons.io.FileUtils
import org.scalatest._

class TransactingRocksDatabasePerformanceSpec extends FlatSpec with KeyValueDatabasePerformanceTrait with ShouldMatchers with BeforeAndAfterEach {
  this: Suite =>

  Storage.initialize()

  var db : KeyValueDatabase = null
  var transactingDB : TransactingRocksDatabase = null

  override def beforeEach() {

    val testPath = new File("./target/unittests-RocksDatabasePerformanceSpec")
    FileUtils.deleteDirectory( testPath )
    transactingDB = new TransactingRocksDatabase( new RocksDatabase( testPath ) )
    transactingDB.beginTransaction()
    db = transactingDB

    super.beforeEach()
  }

  override def afterEach() {
    super.afterEach()

    transactingDB.commitTransaction()
    db.close()
  }
}

File: scalechain-storage/src/test/kotlin/io/scalechain/blockchain/storage/index/TransactingRocksDatabasePerformanceSpec.kt
package io.scalechain.blockchain.storage.index

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import java.io.File

import io.scalechain.blockchain.storage.Storage
import org.junit.runner.RunWith

@RunWith(KTestJUnitRunner::class)
class TransactingRocksDatabasePerformanceSpec : FlatSpec(), Matchers, KeyValueDatabasePerformanceTrait {
  val testPath = File("./target/unittests-DatabasePerformanceSpec")

  lateinit var transactingDB : TransactingKeyValueDatabase
  lateinit override var db : KeyValueDatabase
  lateinit var rocksDB : RocksDatabase

  override fun beforeEach() {
    testPath.deleteRecursively()
    testPath.mkdir()

    rocksDB = RocksDatabase( testPath )
    transactingDB = rocksDB.transacting()
    transactingDB.beginTransaction()
    db = transactingDB

    super.beforeEach()
  }

  override fun afterEach() {
    super.afterEach()

    transactingDB.commitTransaction()
    rocksDB.close()
  }

  init {
    Storage.initialize()
    addTests()
  }
}
